### PR TITLE
Adding DPP-related annotations to OpenAPI spec.

### PR DESCRIPTION
**Pull request by sthesse**
The processing of personal data is key in business processes. Data Protection and Privacy related annotations enable consumers of APIs to identify potential entities and fields with personal data. Based on existing personal data annotations like in ODATA, the PR extends OpenAPI, with DPP annotations. The consumer of APIs can use the proposed DPP annotations to fulfil the requirements from data privacy regulations.

### CODE REVIEW

File: sap-schemas/v3.0/schema.json [hunk 1]
```diff
@@ -718,6 +718,46 @@
...
+          "description": "Primary meaning of the entities in the annotated entity set. Entities annotated with x-sap-dpp-entitySemantics are synonymous to x-sap-dpp-isPotentiallyPersonal.",
+          "enum": [
+            "sap:DataSubject",
+            "sap:DataSubjectDetails",
+            "sap:Other"
+           ]
+        },
+        "x-sap-dpp-dataSubjectRole": {
+          "type": "string",
+          "description": "Role of the data subjects in this set (e.g. employee, customer). Values are application-specific. Can be a static value or a Path expression If the role varies per entity"
```
Comment (ralfhandl) [inline sap-schemas/v3.0/schema.json:733]:
```suggestion
          "description": "Role of the data subjects in this set (e.g. employee, customer). Values are application-specific."
```
OpenAPI does not allow Path expressions, we would have to define how these are expressed if they are needed.

File: sap-schemas/v3.0/schema.json [hunk 2]
```diff
@@ -718,6 +718,46 @@
...
+        },
+        "x-sap-dpp-dataSubjectRoleDescription": {
+          "type": "string",
+          "description": "Language-dependent description of the role of the data subjects in this set (e.g. employee, customer). Values are application-specific. Can be a static value or a Path expression If the role varies per entity"
```
Comment (ralfhandl) [inline sap-schemas/v3.0/schema.json:737]:
```suggestion
          "description": "Language-dependent description of the role of the data subjects in this set (e.g. employee, customer). Values are application-specific."
```

File: sap-schemas/v3.0/schema.json [hunk 3]
```diff
@@ -718,6 +718,46 @@
...
+        },
+        "x-sap-dpp-fieldSemantics": {
+          "type": "string",
+          "description": "Primary meaning of the personal data contained in the annotated property. Changes to values of annotated properties are tracked in the audit log. Use this annotation also on fields that are already marked as contact or address data.Properties annotated with x-sap-dpp-fieldSemantics need not be additionally annotated with x-sap-dpp-isPotentiallyPersonal.",
```
Comment (ralfhandl) [inline sap-schemas/v3.0/schema.json:741]:
```suggestion
          "description": "Primary meaning of the personal data contained in the annotated property. Changes to values of annotated properties are tracked in the audit log. Use this annotation also on fields that are already marked as contact or address data. Properties annotated with x-sap-dpp-fieldSemantics need not be additionally annotated with x-sap-dpp-isPotentiallyPersonal.",
```

File: sap-schemas/v3.0/schema.json [hunk 4]
```diff
@@ -718,6 +718,46 @@
...
+            "sap:ContractRelatedID",
+            "sap:LegalEntityID",
+            "sap:UserID",
+            "sap:EndOfBusinessDate"
+          ]    
+        },
+        "x-sap-dpp-isPotentiallyPersonal": {
+          "type": "boolean",
+          "default": true,
+          "description": "Property contains potentially personal data.Properties annotated with x-sap-dpp-fieldSemantics need not be additionally annotated with this extension."
```
Comment (ralfhandl) [inline sap-schemas/v3.0/schema.json:755]:
```suggestion
          "description": "Property contains potentially personal data. Properties annotated with x-sap-dpp-fieldSemantics need not be additionally annotated with this extension."
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: sap-schemas/v3.0/schema.json
Comment (ralfhandl) [inline sap-schemas/v3.0/schema.json:1]: Comments on this file also apply to the other three files, I just don't like to repeat myself.

File: sap-extensions/v2.0-extensions.json [hunk 5]
```diff
@@ -32,6 +32,49 @@
           "type": "boolean",
           "default": false,
           "description": "Indicates if the marked object represents a root entity. A root entity is the root of a business object. It has a distinct existence. Examples are real-world objects, like a workforce person or a piece of equipment or other more abstract real-world concepts, like a contract or an order. It has an independent lifetime / identity and is globally and uniquely addressable (via the unique identifier provided by the property referenced in x-sap-odm-oid field)."
+        },
+        "x-sap-dpp-entitySemantics": {
```
Comment (pavelkornev) [inline sap-extensions/v2.0-extensions.json:36]: We have a convention for extension fields names that they must follow kebab-case-notation.

There are some old fields with camelCase notation, those are exceptions due to historical reasons.

**Review by pavelkornev — CHANGES_REQUESTED** [review]

**Comment by pavelkornev** [comment]
@sthesse aren't fields the same for OpenAPI v2 and OpenAPI v3? If so, please place them in https://github.com/SAP/openapi-specification/blob/main/sap-extensions/extensions.json, no need to duplicate content.

**Review by pavelkornev — APPROVED** [review]

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**